ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py

```diff
@@ -268,6 +268,29 @@ def _local_path_groups_connected_to_pathfinder(self) -> list:
             if any(wan_interface["connected_to_pathfinder"] for wan_interface in path_group["interfaces"])
         ]
 
+    def _filter_path_groups_for_auto_lb_policy(self, path_groups: list) -> list:
+        """
+        Take a list of path_groups and filter them based on if they are excluded or not
+        """
+        filtered_path_groups = {}
+        for path_group_name in path_groups:
+            path_group = get_item(
+                self.shared_utils.wan_path_groups,
+                "name",
+                path_group_name,
+                required=True,
+            ).copy()
+
+            default_preference = get(path_group, "default_preference", 1)
+            context_path = f"wan_path_groups[{path_group_name}]"
+            priority = self._path_group_preference_to_eos_priority(default_preference, context_path)
+            if not path_group.get("exclude_from_auto_lb_policies"):
+                if priority not in filtered_path_groups:
+                    filtered_path_groups[priority] = []
+                filtered_path_groups[priority].append(path_group_name)
+
+        return [{"names": names, "preference": priority} for priority, names in filtered_path_groups.items()]
```

Comment:
refactor this and see if this is really needed - maybe can just include default preference in `self.shared_utils.wan_local_path_groups` instead so that you can use it when required in the existing functions

Reply:
This looks good to me, `wan_local_path_groups` has `default_preference` field if you want to fetch it. 

The question is wether you want to use this default_preference for manual policies as well ( but without preference specified ) or just for auto generated policies.

If just for auto generated policies this seems like a good place. 

I thought that for an auto policy we will have a list of path groups, we call this function, which gives us policy and then `_generate_wan_load_balance_policy` takes that policy.

The other option is to have `_generate_wan_load_balance_policy` take some parameter, if the policy is auto or not, and then pick up the preference from wan_local_path_groups, and then do magic, but because the input parameter is policy for it, it won't look clean, as now even within a policy row, you will need to check preference.

Reply:
> The question is wether you want to use this default_preference for manual policies as well ( but without preference specified ) or just for auto generated policies.

That was not the idea, the idea is that by default we would be using `1` (`preferred`) but thinking further about this it could actually make sense to use the default policy set under wan_path_groups. @ClausHolbechArista any opinion?

Even if this is for autogenerated policies, is it not possible to avoid looping though all the path_groups  here but rather using watever is is in `wan_local_path_groups` when the automatic policy is being used?

Reply:
```
That was not the idea, the idea is that by default we would be using 1 (preferred) but thinking further about this it could actually make sense to use the default policy set under wan_path_groups. @ClausHolbechArista any opinion?
```
This doesn't properly suit the inputs structure, 
as someone could in policy specify 
- names : [P1, P2]
----
where P1 has default pref of 1 and P2 has 2, we can obviously now throw error, but seems confusing behaviour.

> Even if this is for autogenerated policies, is it not possible to avoid looping though all the path_groups here but rather using watever is is in wan_local_path_groups when the automatic policy is being used?
That is what we are doing, no ? it's a helper function which iterates over path groups which are to be used in load balance policy and then reads the priority from wan_local_path_groups, and returns.

Doing it in `_generate_wan_load_balance_policy ` will increase code complexity further.

Reply:
> This doesn't properly suit the inputs structure,
> as someone could in policy specify
> 
> names : [P1, P2]
> where P1 has default pref of 1 and P2 has 2, we can obviously now throw error, but seems confusing behaviour.

good point so probably a bad idea for non automatically generated policies

Reply:
Refactored as discussed.

Reply:
@ayushmittal-arista - we discussed with @ClausHolbechArista and actually we are coming back on this

>The question is wether you want to use this default_preference for manual policies as well ( but without preference specified ) or just for auto generated policies.

Let's implement this also in this PR and detail that if you don;t specify a preference for an entry, each path-group in the list will pick the default preference you have implemented (and yes, they may not be the same)